bootstrap/handlers/httpserver.go

```diff
@@ -156,7 +168,70 @@ func (b *HttpServer) BootstrapHandler(
 		}()
 
 		b.isRunning = true
-		err := server.ListenAndServe()
+
+		var ln net.Listener
+		switch bootstrapConfig.Service.SecurityOptions["Mode"] {
+		case "zerotrust":
+			secretProvider := container.SecretProviderExtFrom(dic.Get)
+			var zitiCtx ziti.Context
+			var ctxErr error
+			if secretProvider != nil {
+				jwt, jwtErr := secretProvider.GetSelfJWT()
+				if jwtErr != nil {
+					lc.Errorf("could not load jwt: %v", jwtErr)
+				}
+				lc.Info("using zerotrust - look at you go: " + jwt)
+				openZitiRootUrl := "https://" + bootstrapConfig.Service.SecurityOptions["OpenZitiController"]
+				caPool, caErr := ziti.GetControllerWellKnownCaPool(openZitiRootUrl)
+				if caErr != nil {
+					panic(caErr)
+				}
+
+				credentials := edge_apis.NewJwtCredentials(jwt)
+				credentials.CaPool = caPool
+
+				cfg := &ziti.Config{
+					ZtAPI:       openZitiRootUrl + "/edge/client/v1",
+					Credentials: credentials,
+				}
+				cfg.ConfigTypes = append(cfg.ConfigTypes, "all")
+
+				zitiCtx, ctxErr = ziti.NewContext(cfg)
+				if ctxErr != nil {
+					panic(ctxErr)
+				}
+			} else {
+				ozIdFile := bootstrapConfig.Service.SecurityOptions["OpenZitiIdentityFile"]
+				if strings.TrimSpace(ozIdFile) == "" {
+					panic("here?")
+				} else {
+					zitiCtx, ctxErr = ziti.LoadContext(ozIdFile)
+					if ctxErr != nil {
+						panic(ctxErr)
+					}
+				}
+
+				ziti.DefaultCollection.Add(zitiCtx)
+			}
+
+			serviceName := bootstrapConfig.Service.SecurityOptions["OpenZitiServiceName"]
+			ln, err = zitiCtx.Listen(serviceName)
+			if err != nil {
+				panic("could not bind service " + serviceName + ": " + err.Error())
+			}
+
+			zc.c = &zitiCtx
+		case "http":
+		default:
+			lc.Warnf("using ListenMode 'http' at %s", addr)
+			ln, err = net.Listen("tcp", addr)
+		}
+		server.ConnContext = mutator
+
+		if err == nil {
```

Comment:
Check not needed if properly log err and return `false` above.

Reply:
moved the server.serve to within both if's instead and adopted the existing pattern of not returning false, but instead setting an error to get through to the `if err != nil && err != http.ErrServerClosed {` block